### DISCUSSION TITLE
Constructor signature for std::make_unique

### DISCUSSION THREAD

**Discussion by ytimenkov** [microsoft/vscode-cpptools#8066] [Show and tell]
This is awesome:
![image](https://user-images.githubusercontent.com/7413219/131119970-1e49d9bb-b4bf-400d-85e5-6752be6a3df4.png)

Somehow I missed it in release notes... Can it do similar tricks for other forwarding functions or just `std::make_*` ?

**Comment by sean-mcmanus** [comment]
It doesn't seem to work with a user-defined make_* function that uses std::forward. I'm not sure yet what is special about that handling, but I'll see if I can find out...

**Reply by ytimenkov** [reply]
Regardless of what you find out I think this feature itself deserves mentioning in release notes or so.

I understand that you're not directly in control of the engine, but a lot of nice things pop up there which are discovered by accident. I wonder how much more I'm not aware of 🤭

**Reply by sean-mcmanus** [reply]
The perfect forwarding is intentionally limited to std::make_unique, make_shared, emplace, emplace_back...I'm not sure if they'll make that more general later on.

We got the feature from VS in our 1.2 release 7 months ago and it was blogged about at https://devblogs.microsoft.com/cppblog/intellisense-improvements-in-visual-studio-2019/#perfect-forwarding-intellisense-make_unique-make_shared-emplace-emplace_back . What "release notes" were you referring to? We generally don't re-blog about features that are shared between VS and VS Code. The VS Code release notes are for VS Code stuff and doesn't include updates for extensions, and we don't ship a "release notes", but we do have a changelog, which usually just mentions changes that are being tracked on our GitHub page and not 100% of the changes (i.e. changes such as this one that come from our shared VS components are usually not included, unless we somehow notice that it's worth mentioning at the time, which we didn't in this case).

UPDATE: Hopefully, in the future, the blog-able VS updates that affect VS Code can be added to the VS Code blog and changelog too (assuming the VS team communicates the changes to our team)...although in this case, the VS blog was released a month after the feature shipped in VS Code, so it's possible we might still miss some blog-worthy IntelliSense updates, unless we update our blog/changelog later on.
